app/src/main/java/hu/vmiklos/plees_tracker/MainActivity.kt

```diff
@@ -123,6 +123,12 @@ class MainActivity : AppCompatActivity() {
         return null
     }
 
+    private fun setDashboardText(s: String) {
+        val index = resources.getStringArray(R.array.duration_entry_values).indexOf(s)
+        val durationStr = resources.getStringArray(R.array.duration_entries)[index]
+        findViewById<TextView>(R.id.dashboard_header)?.text = getString(R.string.dashboard, durationStr)
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
```

Comment:
I've marked this PR as draft because I wasn't able to figure out a good way to have the setting the duration update on the main activity when changed. One option I thought of was to implement a `onResume`, but things got messy pretty quickly because `onCreate` does a lot. Any guidance here would be much appreciated.

Reply:
All sleeps and stats are already updated using an observer pattern, i.e. when a new sleep is added, then Room notifies us, we don't have to update any sleep or stat explicitly. Would it be possible to build on top of this? Either extract the Room callback, so you can invoke it when the pref changes, or you could trigger a Room invalidation. Additionally, you  probably want to use a preference change listener, so pref change triggers the observer callback. Does that help?